openstack_controller/tests/test_unit_cinder.py

```diff
@@ -211,3 +213,167 @@ def test_response_time_volumev3(aggregator, check, dd_run_check, mock_http_get):
+        ),
+    ],
+)
+@pytest.mark.usefixtures('mock_http_get', 'mock_http_post', 'openstack_connection')
+def test_block_storage_metrics(aggregator, check, dd_run_check):
+    dd_run_check(check)
+    aggregator.assert_metric(
+        'openstack.cinder.volume.count',
+        count=1,
+        value=1,
+        tags=[
+            'domain_id:default',
+            'project_name:demo',
+            'project_id:1e6e233e637d4d55a50a62b63398ad15',
+            'volume_id:9c762008-d70f-44d1-af02-98e1da79ee4b',
+            'volume_size:1',
+            'volume_name:',
+            'volume_status:in-use',
+            'keystone_server:http://127.0.0.1:8080/identity',
+        ],
+    )
+    aggregator.assert_metric(
+        'openstack.cinder.volume.count',
+        count=1,
+        value=1,
+        tags=[
+            'domain_id:default',
+            'project_name:demo',
+            'project_id:1e6e233e637d4d55a50a62b63398ad15',
+            'volume_id:259b16de-727f-4011-8388-84d17a9ae594',
+            'volume_size:1',
+            'volume_name:',
+            'volume_status:in-use',
+            'keystone_server:http://127.0.0.1:8080/identity',
+        ],
+    )
+
+
+@pytest.mark.parametrize(
+    ('instance', 'paginated_limit', 'api_type', 'expected_api_calls'),
+    [
+        pytest.param(
+            configs.REST,
+            1,
+            ApiType.REST,
+            1,
+            id='api rest low limit',
+        ),
+        pytest.param(
+            configs.REST,
+            1000,
+            ApiType.REST,
+            1,
+            id='api rest high limit',
+        ),
+        pytest.param(
+            configs.SDK,
+            1,
+            ApiType.SDK,
+            1,
+            id='api sdk low limit',
+        ),
+        pytest.param(
+            configs.SDK,
+            1000,
+            ApiType.SDK,
+            1,
+            id='api sdk high limit',
+        ),
+    ],
+)
+@pytest.mark.usefixtures('mock_http_get', 'mock_http_post', 'openstack_connection')
+def test_block_storage_pagination(
+    aggregator,
+    instance,
+    openstack_controller_check,
+    paginated_limit,
+    expected_api_calls,
+    api_type,
+    dd_run_check,
+    connection_block_storage,
+    mock_http_get,
+):
+    paginated_instance = copy.deepcopy(instance)
+    paginated_instance['paginated_limit'] = paginated_limit
+    dd_run_check(openstack_controller_check(paginated_instance))
+    if api_type == ApiType.REST:
+        args_list = []
+        for call in mock_http_get.call_args_list:
+            args, kwargs = call
+            args_list += list(args)
+            params = kwargs.get('params', {})
+            limit = params.get('limit')
+            args_list += [(args[0], limit)]
+        assert (
+            args_list.count(
+                ('http://127.0.0.1:8776/volume/v3/1e6e233e637d4d55a50a62b63398ad15/volumes/detail', paginated_limit)
+            )
+            == 2
+            if paginated_limit == 1
```

Comment:
could you make this a parameter (ie have `expected_api_calls_proj_1` and `expected_api_calls_proj_2`) or something similar?